### ISSUE TITLE
Remove the usage of Micronaut HTTP client in plugins

### ISSUE THREAD

**Issue by loicmathieu** [kestra-io/kestra#3279]
### Issue description

Some plugins uses the Micronaut HTTP Client.
During migration to Micronaut 4, we had a lot of issue due to the Plugin classloader and the usage, in the HTTP Client of service loader.
There is now a good HTTP Client directly in the JDK and we should avoid depending on Micronaut in plugins as much as possible.

During the process, great care should be done for JSON support, we should idealy use the same JSON Mapper that we use in Kestra itself to avoid different behavior, be careful also that all Jackson modules should work to avoid hard to catch issue like https://github.com/kestra-io/plugin-dbt/issues/98

**Comment by anna-geller** [comment]
acc. to Ludo: 
1. Jira / hugging face & vault secret are still affected, the rest is already done
2. we need to cover it in 1.0, as it's a major source of bugs, and only a few minutes change

**Comment by Ben8t** [comment]
Once https://github.com/kestra-io/secret-vault/pull/42 is merged, the issue can be close thanks to Malay's work 👍

**Comment by Malaydewangan09** [comment]
closed by 
- https://github.com/kestra-io/plugin-huggingface/pull/15
- https://github.com/kestra-io/plugin-jira/pull/26
- https://github.com/kestra-io/secret-vault/pull/42
